server/src/test/omtLinkProvider.test.ts

```diff
@@ -0,0 +1,132 @@
+import { expect, use } from 'chai';
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+import { stub } from 'sinon';
+import { DocumentLink, Position, Range } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import OMTLinkProvider from '../omtLinkProvider';
+import { WorkspaceLookup } from '../workspaceLookup';
+import * as sinonChai from 'sinon-chai';
+use(sinonChai);
+
+type Case = {
+    i: number,
+    line: number,
+    start: number,
+    end: number,
+    should: string,
+    target: string | undefined,
+    module?: string,
+};
+
+describe('OMTLinkProvider', () => {
+    let linkProvider: OMTLinkProvider;
+    let lookupStub: WorkspaceLookup;
+
+    beforeEach(() => {
+        lookupStub = <WorkspaceLookup>{
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            getModulePath: (_module: string) => '',
+        };
+
+        linkProvider = new OMTLinkProvider(lookupStub);
+    });
+
+    describe('provideDocumentLinks', () => {
+        let actualDocumentLinks: DocumentLink[];
+
+        beforeEach(() => {
+            const uri = 'testFixture/one/imports.omt';
+            const textDocument = TextDocument.create(uri, 'omt', 1, readFileSync(resolve(uri)).toString());
+            const errorStub = stub(console, 'error');
+            actualDocumentLinks = linkProvider.provideDocumentLinks(textDocument);
+            //- called in the catch clause for reading tsconfig-invalid
+            expect(errorStub).to.be.calledOnce;
+            errorStub.restore();
+        })
+
+        // test uri links with and without shorthands
+        const cases: Case[] = [
+            { i: 0, line: 1, start: 1, end: 18, target: undefined, should: 'should make a link for a declared import', module: 'Declared' },
+            { i: 1, line: 3, start: 5, end: 20, target: resolve('testFixture/relative.omt'), should: 'should make a link for a relative path' },
+            { i: 2, line: 5, start: 4, end: 24, target: resolve('testFixture/one/shorthanded.omt'), should: 'should recognize a shorthand from the tsconfig in current project folder' },
+            { i: 3, line: 7, start: 5, end: 29, target: resolve('testFixture/one/quotedShorthand.omt'), should: 'should ignore quotes 1' },
+            { i: 4, line: 9, start: 5, end: 35, target: resolve('testFixture/one/doubleQuotedShorthand.omt'), should: 'should ignore quotes 2' },
+            { i: 5, line: 11, start: 4, end: 31, target: resolve('testFixture/one/@two/siblingShorthanded.omt'), should: 'should not ignore a shorthand from a sibling path' },
+            { i: 6, line: 13, start: 4, end: 27, target: resolve('testFixture/three/parentPathed.omt'), should: 'should recognize a shorthand from the tsconfig in a parent project' },
+        ];
+
+        // test all cases
+        cases.forEach((value: Case) => {
```

Comment:
It's not wrong to type `value` here, but typescript should be able to figure that out on its own, because `cases` is an array of `Case`s.

Reply:
done, it used to do something. not anymore